### ISSUE TITLE
Adding Unit Test Cases using Jest Testing Library

### ISSUE THREAD

**Issue by Jagroop2000** [adarshanand67/ShareMe#28]
**Subject:** Request for Jest integration in ReactJS projects on GitHub
**Hi there !!**

I am writing to request the **integration of Jest**, a popular JavaScript testing library, into your project that uses ReactJS. This would allow developers to write and run unit tests for their React components directly on the platform. Jest's simplicity and power would greatly improve the development experience and increase the reliability of code.

And the good thing is Jest testing files wouldn't add any memory consumption to the project when we make a build or it is on production.
Please add **hard level label** under **JWoC** to this or you can decide it while merging or reviewing PR request.
Thank you for considering this request. I look forward to your response.

Best regards,
Jagroop

**Comment by adarshanand67** [comment]
Alright Start working on this @Jagroop2000

**Comment by Jagroop2000** [comment]
Hi @adarshanand67 ,
I have checked that this project is migrated to Vite.
So here JEST will cause issue as for Vite there Vitest.
Can I go with Vitest ? Please confirm.

**Comment by adarshanand67** [comment]
> Hi @adarshanand67 ,
> I have checked that this project is migrated to Vite.
> So here JEST will cause issue as for Vite there Vitest.
> Can I go with Vitest ? Please confirm.

Yes good observation, I made this issue when project was with CRA. You can proceed with Vitest
